src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -1870,16 +1900,15 @@ public EntryResult tMedicationActivity2MedicationStatement(MedicationActivity cd
 		}
 
 		// consumable.manufacturedProduct -> medication
+
 		if (cdaMedicationActivity.getConsumable() != null && !cdaMedicationActivity.getConsumable().isSetNullFlavor()) {
-			if (cdaMedicationActivity.getConsumable().getManufacturedProduct() != null
-					&& !cdaMedicationActivity.getConsumable().getManufacturedProduct().isSetNullFlavor()) {
-				Bundle fhirMedicationBundle = tManufacturedProduct2Medication(
-						cdaMedicationActivity.getConsumable().getManufacturedProduct());
-				for (BundleEntryComponent entry : fhirMedicationBundle.getEntry()) {
-					result.addResource(entry.getResource());
-					if (entry.getResource() instanceof org.hl7.fhir.dstu3.model.Medication) {
+			EntryResult fhirMedicationResult = tManufacturedProduct2Medication(cdaMedicationActivity.getConsumable());
+			if (fhirMedicationResult.hasResult()) {
+				for (BundleEntryComponent entry : fhirMedicationResult.getBundle().getEntry()) {
+					if (entry.getResource().fhirType().contentEquals("Medication")) {
```

Comment:
why did this change?

Reply:
Afsin recommended that I consolidate some repetitive code, and at first I thought this part was shared between this section and another section that transforms a ManufacturedProduct to a Medication. In the end it wasn't that similar but overloading the function made it a bit smaller in both sections.... I didn't intend to add the "contentEquals" though. I'm going to go back and change that.

Reply:
I was also changing the output to EntryResult since we're using that more than Bundle, but maybe that's only for top-level fhir resources?